compiler/rustc_data_structures/src/graph/scc/mod.rs

```diff
@@ -48,6 +48,20 @@ pub trait Annotation: Debug + Copy {
     }
 }
 
+/// An accumulator for annotations.
+pub trait Annotations<N: Idx, S: Idx, A: Annotation> {
+    fn new(&self, element: N) -> A;
+    fn annotate_scc(&mut self, scc: S, annotation: A);
+}
+
+/// The nil annotation accumulator, which does nothing.
+impl<N: Idx, S: Idx> Annotations<N, S, ()> for () {
+    fn new(&self, _element: N) -> () {
+        ()
+    }
```

Comment:
:<

Reply:
i wrote this because it should just be `fn new(&self, _element: N) {}`

Reply:
Haha! I forgot () is actually the default no return value value